### ISSUE TITLE
Roadmap

### ISSUE THREAD

**Issue by sda030** [BroVic/RQDAassist#14]
Hi Victor,
I haven never used RQDA myself (and I am not a qualitative researcher), but as part of a research application for developing some R packages I am considering including RQDA2 as a work package because good colleagues want this package revived (and I want more colleagues to use R rather than expensive software). 
I understand that the main reasons why RQDA fails to install from CRAN is that dependencies are archived due to lack of maintenance (RGtk2, gWidgetsRGtk2, gWidgets, cairoDevice). Seems they are all professor Lawrence's packages, and seems hard to get a hold of him. 
You have made RQDAassist to circumvent some of the issues, but it is obviously not the best solution.
1) Could you help me make a detailed bullet point list of all the challenges remaining to get RQDA to get to work perfectly? And perhaps an estimate of how difficult/how much work it would take? It would then be possible for me to see what can be funded.
2) Also, have you gained sufficient programming knowledge to handle some of the work?
3) I am also wondering whether Tcl/tk-GUIs will be too fragile for the future, and quite complicated to maintain. Few nowadays know Tcl/tk. Do you think Shiny could be an alternative?

Stephan

**Comment by BroVic** [comment]
Hello Stephan.

I initially wrote this package to meet a personal and team-level need, and as a quick fix. I've nursed the idea of working on a more sustainable solution but I recall seeing a post by the author of RGtk2, where he stated his intention to contact CRAN regarding the archival of the package. So I've been waiting to see if there would be any change in the status quo. 

I can enumerate a few challenges facing RQDA:
- Upgrade of gWidgets and gWidgetRGtk2 to gWidgets2 and gWidgets2RGtk2,, respectively. The RQDA repo (also a fork developed by Jan Marvin) made changes that depended on the new packages. 
- Archiving of RGtk2: The package was categorized as "Orphaned" and subsequently archived on CRAN. I don't recall why, as there are several packages that depended on it. 
-  Shift to Universal C Runtime (UCRT) in R 4.2: I froze this package to work for R versions lower than 4.2. This is because,    the current version of RGtk2, cannot be built from source using UCRT. 
- RGtk2 dependencies: I've also noticed the GTK distribution has been upgraded to version 3 (when I last checked. This is a dependency of RGtk2.

These are the immediate issues I see (though I may have missed something), and have not done much about it for want of time. I believe these are what needs to be addressed for a robust solution to the RQDA conundrum.

I don't mind being part of an effort to revive the package. Apart from R, I can also code in C and C++. And for performance reasons, I don't think Shiny would be a great approach. 

Cheers,

Victor
